### PR TITLE
REM-1109: change account transferring with node account condition

### PR DESCRIPTION
**Pull request by dmytrostriletskyi**
### Jira references
- Ticket (task) — https://remmeio.atlassian.net/browse/REM-1109

### CODE REVIEW

File: remme/tp/account.py [hunk 1]
```diff
@@ -106,36 +121,40 @@ def _transfer_from_address(self, context, address_from, transfer_payload):
...
                 and transfer_payload.address_to != ZERO_ADDRESS:
             raise InvalidTransaction('Receiver address has to be of an account type.')
 
         if address_from == transfer_payload.address_to:
             raise InvalidTransaction('Account cannot send tokens to itself.')
 
-        signer_account, receiver_account = get_multiple_data(context, [
-            (address_from, Account),
-            (transfer_payload.address_to, Account),
+        sender_account_pb_class = Account if address_from[:6] == self._prefix else NodeAccount
```
Comment (eugene-babichenko) [inline remme/tp/account.py]: Basically this allows as to set any prefix different from `self._prefix`. That is pretty much enough to break the state.

File: remme/tp/account.py [hunk 2]
```diff
@@ -97,7 +108,11 @@ def _transfer(self, context, public_key, transfer_payload):
         References:
             - https://github.com/Remmeauth/remme-core/blob/dev/remme/genesis/__main__.py
         """
-        address = self.make_address_from_data(public_key)
+        if transfer_payload.sender_account_type == TransferPayload.SenderAccountType.Value('ACCOUNT'):
+            address = self.make_address_from_data(public_key)
+
+        else:  # sender account type is node account
```
Comment (eugene-babichenko) [inline remme/tp/account.py]: This looks looks like it allows to pass anything different from `TransferPayload.SenderAccountType.Value('ACCOUNT')`.

File: remme/tp/account.py [hunk 3]
```diff
@@ -97,7 +108,11 @@ def _transfer(self, context, public_key, transfer_payload):
         References:
             - https://github.com/Remmeauth/remme-core/blob/dev/remme/genesis/__main__.py
         """
-        address = self.make_address_from_data(public_key)
+        if transfer_payload.sender_account_type == TransferPayload.SenderAccountType.Value('ACCOUNT'):
```
Comment (eugene-babichenko) [inline remme/tp/account.py]: You can just use `TransferPayload.ACCOUNT` here.

File: remme/tp/account.py (hunk 2 above)
Comment (dmytrostriletskyi) [reply remme/tp/account.py]: @eugene-babichenko, we have a [test](https://github.com/Remmeauth/remme-core/blob/aa3e42e67e833ec9dd8f9a4632a774474ddc880d/testing/unit/tp/account/test_transfer.py#L124) for checking sender account type value.

File: remme/tp/account.py (hunk 3 above)
Comment (dmytrostriletskyi) [reply remme/tp/account.py]: @eugene-babichenko, done.

File: remme/tp/account.py (hunk 1 above)
Comment (dmytrostriletskyi) [reply remme/tp/account.py]: @eugene-babichenko, done.

File: remme/tp/account.py [hunk 4]
```diff
@@ -97,45 +96,73 @@ def _transfer(self, context, public_key, transfer_payload):
...
+        else:
+            address = NodeAccountHandler().make_address_from_data(public_key)
 
         return self._transfer_from_address(context, address, transfer_payload)
 
+    def is_address_account_type(self, address):
+        """
+        Check if address is account address type.
+        """
+        if address.startswith(self._prefix):
```
Comment (eugene-babichenko) [inline remme/tp/account.py]: Why not just `return a or b or c`?

Comment (dmytrostriletskyi) [reply remme/tp/account.py]: @eugene-babichenko, didn't get you, can you rephase?

Comment (eugene-babichenko) [reply remme/tp/account.py]: Particularly

```python
def is_address_account_type(self, address):
    return address.startswith(self._prefix) or \
        address.startswith(NodeAccountHandler()._prefix) or \
        address == ZERO_ADDRESS
```

Comment (dmytrostriletskyi) [reply remme/tp/account.py]: @eugene-babichenko, looks a bit ugly that mine approach, imho. But done.

**Pull request merged**